panther_analysis_tool/detection.py

```diff
@@ -0,0 +1,191 @@
+"""
+Panther Analysis Tool is a command line interface for writing,
+testing, and packaging policies/rules.
+Copyright (C) 2020 Panther Labs Inc
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import logging
+import re
+import traceback
+from abc import abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from types import ModuleType
+from typing import List, Optional
+
+from panther_analysis_tool.util import id_to_path, import_file_as_module, store_modules
+
+# Temporary alias for compatibility
+get_logger = logging.getLogger
+
+
+# pylint: disable=too-many-instance-attributes,unsubscriptable-object
+@dataclass
+class DetectionResult:
+    """Class containing the result of running a detection"""
+
+    detection_id: str
+    detection_severity: str
+    detection_type: Optional[str] = None
```

Comment:
This field should probably be required, but I would advise to not do it now, because it will trigger another round of changes in tests 😄 .